### PR TITLE
chore: add s_subIds collection to SubscriptionAPI.sol

### PR DESCRIPTION
**Pull request by makramkd**
Add an `EnumerableSet` to the V2+ coordinator in order to fetch all subscription id's that are active on the contract.

This is so that chain readers can paginate through all active VRF subscriptions without having to scrape event history, which may or may not be available depending on the node being connected to.

### CODE REVIEW

File: contracts/src/v0.8/dev/vrf/SubscriptionAPI.sol [hunk 1]
```diff
@@ -355,6 +389,7 @@ abstract contract SubscriptionAPI is ConfirmedOwner, IERC677Receiver, IVRFSubscr
     }
     delete s_subscriptionConfigs[subId];
     delete s_subscriptions[subId];
+    s_subIds.remove(subId);
```
[the remark this answers is not in the document]
Comment (makramkd) [reply contracts/src/v0.8/dev/vrf/SubscriptionAPI.sol:392]: According to the OZ docstring this is an O(1) operation. I think under the hood it's just a mapping deletion.

File: contracts/test/v0.8/foundry/vrf/VRFV2Plus.t.sol [hunk 2]
```diff
@@ -141,6 +142,62 @@ contract VRFV2Plus is BaseTest {
...
+    for (uint batchIdx = 0; batchIdx < subIds.length; batchIdx++) {
+      for (uint subIdx = 0; subIdx < subIds[batchIdx].length; subIdx++) {
+        s_testCoordinator.getSubscription(subIds[batchIdx][subIdx]);
+        actualNumSubs++;
+      }
+    }
+    assertEq(actualNumSubs, s_testCoordinator.getActiveSubscriptionIdsLength());
+
+    {
+      // cancel a bunch of subscriptions, assert that they are not returned
```
[the remark this answers is not in the document]
Comment (makramkd) [reply contracts/test/v0.8/foundry/vrf/VRFV2Plus.t.sol:168]: I believe it's happening (I see it printed out in the `-vvvv` logs) but you're right, prob better to add that assertion.

File: contracts/test/v0.8/foundry/vrf/VRFV2Plus.t.sol [hunk 3]
```diff
@@ -452,4 +459,8 @@ contract VRFV2Plus is BaseTest {
     (uint96 linkBalanceAfter, , , , ) = s_testCoordinator.getSubscription(subId);
     assertApproxEqAbs(linkBalanceAfter, linkBalanceBefore - 280_000, 20_000);
   }
+
+  function testMyFunction() public {
```
Comment (jinhoonbang) [inline contracts/test/v0.8/foundry/vrf/VRFV2Plus.t.sol:463]: *nit. remove?

**Review by jinhoonbang — APPROVED** [review]

File: contracts/test/v0.8/foundry/vrf/VRFV2Plus.t.sol (hunk 3 above)
Comment (makramkd) [reply contracts/test/v0.8/foundry/vrf/VRFV2Plus.t.sol:463]: Yikes, good find. Committed this by mistake.

**Review by kidambisrinivas — APPROVED** [review]
LGTM

**Pull request merged**